### PR TITLE
Setup classes

### PR DESCRIPTION
**Pull request by PascalSCoder**
Seperated the flow for socket recv into multiple classes/functions. Also added basic functionality to receive data in multiple recv calls.

### CODE REVIEW

File: srcs/Webserv.cpp [hunk 1]
```diff
@@ -62,69 +62,85 @@ void	Webserv::setupSocket()
...
+{
+	int fdSize;
+
+	fdSize = _fds.size();
+	for (int i = 1; i < fdSize; ++i) // i = 1, because we don't need to check the listening socket
+	{
+		std::cout << "Revents : " << _fds[i].revents << std::endl;
+		if (_fds[i].revents == 0)
+			continue ;
+
```
Comment (Ryno95) [inline srcs/Webserv.cpp:75]: When I run this and cancel disconnect the client it goes into an infinite loop, maybe a linux mac thing? 
Does this work on Linux (I'm assuming that's wha you are using at home)? Is that why this block of code is commented out?

**Review by Ryno95 — CHANGES_REQUESTED** [review]
Client needs to be added to CLASSES in the Makefile.
Killer readable and clear code :)

File: srcs/Webserv.cpp (hunk 1 above)
Comment (PascalSCoder) [reply srcs/Webserv.cpp:75]: Yap, correct!

**Review by Ryno95 — APPROVED** [review]
Nois!

**Pull request merged**